testsuite/integration-arquillian/servers/auth-server/services/testsuite-providers/src/main/java/org/keycloak/testsuite/services/clientpolicy/executor/TestDPoPSimulateExecutor.java

```diff
@@ -0,0 +1,102 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.testsuite.services.clientpolicy.executor;
+
+import java.security.PublicKey;
+
+import org.jboss.logging.Logger;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.ProtocolMapperModel;
+import org.keycloak.protocol.oidc.mappers.HardcodedClaim;
+import org.keycloak.protocol.oidc.mappers.OIDCAttributeMapperHelper;
+import org.keycloak.representations.idm.ClientPolicyExecutorConfigurationRepresentation;
+import org.keycloak.services.clientpolicy.ClientPolicyContext;
+import org.keycloak.services.clientpolicy.ClientPolicyEvent;
+import org.keycloak.services.clientpolicy.ClientPolicyException;
+import org.keycloak.services.clientpolicy.context.TokenResponseContext;
+import org.keycloak.services.clientpolicy.executor.ClientPolicyExecutorProvider;
+
+/**
+ * @author <a href="mailto:takashi.norimatsu.ws@hitachi.com">Takashi Norimatsu</a>
+ */
+public class TestDPoPSimulateExecutor implements ClientPolicyExecutorProvider<ClientPolicyExecutorConfigurationRepresentation> {
+
+    public static final String DUMMY_JKT = "0ZcOCORZNYy-DWpqq30jZyJGHTN0d2HglBV3uiguA4I";
+
+    private static final Logger logger = Logger.getLogger(TestDPoPSimulateExecutor.class);
+
+    protected final KeycloakSession session;
+
+    private PublicKey DPoPPublicKey;
+
+    public TestDPoPSimulateExecutor(KeycloakSession session) {
+        this.session = session;
+    }
+
+    @Override
+    public String getProviderId() {
+        return TestDPoPSimulateExecutorFactory.PROVIDER_ID;
+    }
+
+    @Override
+    public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyException {
+        ClientPolicyEvent event = context.getEvent();
+        logger.tracev("Client Policy Trigger Event = {0}",  event);
+        switch (event) {
+            case TOKEN_RESPONSE:
+                verifyDPoPProof(context);
+                bindTokensWithDPoPPublicKey((TokenResponseContext)context);
+                break;
+            case TOKEN_REFRESH_RESPONSE:
+                verifyDPoPProof(context);
+                verifyBindingTokensWithDPoPPublicKey((TokenResponseContext)context);
+                bindTokensWithDPoPPublicKey((TokenResponseContext)context);
+                break;
+            default :
+                return;
+        }
+    }
+
+    private void verifyDPoPProof(ClientPolicyContext context) throws ClientPolicyException {
+        // retrieve a DPoP Proof JWT from the "DPoP" HTTP header
```

Comment:
Question: Should there be follow-up issue to uncomment (and test) this once DPoP is available in Keycloak? Or possibly it can be done during DPoP initial PR itself? In this case, note should be added likely to the DPoP issue about that?

Reply:
As you suggested, I think this test should be done in the follow-up issue of the DPoP initial PR. Therefore, I will remove this test from the PR.